src/common/checksums.cpp

```diff
@@ -143,14 +123,19 @@ QByteArray findBestChecksum(const QByteArray &_checksums)
     if (_checksums.isEmpty()) {
         return {};
     }
-    const auto checksums = QString::fromUtf8(_checksums);
-    int i = 0;
+
     // The order of the searches here defines the preference ordering.
-    if (-1 != (i = checksums.indexOf(QLatin1String("SHA3-256:"), 0, Qt::CaseInsensitive))
-        || -1 != (i = checksums.indexOf(QLatin1String("SHA256:"), 0, Qt::CaseInsensitive))
-        || -1 != (i = checksums.indexOf(QLatin1String("SHA1:"), 0, Qt::CaseInsensitive))
-        || -1 != (i = checksums.indexOf(QLatin1String("MD5:"), 0, Qt::CaseInsensitive))
-        || -1 != (i = checksums.indexOf(QLatin1String("ADLER32:"), 0, Qt::CaseInsensitive))) {
+    // usually not a good idea to use toUpper on a byte array but we only care for the id which has no special chars
```

Comment:
Sounds like you really want to have an assertion there. Assumptions need to be verified.